ibm/mas_devops/roles/cp4d_service/tasks/determine-storage-classes.yml

```diff
@@ -21,19 +21,37 @@
 # -----------------------------------------------------------------------------
 - name: Default Metadata Storage if not set by user
   when:
-    - cpd_service_name != "wd"
+    - cpd_service_name != "wd" and cpd_service_name != "ca"
     - cpd_service_storage_class is not defined or cpd_service_storage_class == ""
   set_fact:
     cpd_service_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.defaultStorageClass(default_storage_classes_rwx) }}"
 
-- name: Default Metadata Storage if not set by user (Watson Discovery only)
+- name: Default Metadata Storage if not set by user (Watson Discovery)
   when:
     - cpd_service_name == "wd"
     - cpd_service_storage_class is not defined or cpd_service_storage_class == ""
     - cpd_service_block_storage_class is not defined or cpd_service_block_storage_class == ""
   set_fact:
     cpd_service_storage_class: "{{ (cpd_minor_version == '4.6') | ternary(lookup_storageclasses | ibm.mas_devops.defaultStorageClass(default_storage_classes_rwx), lookup_storageclasses | ibm.mas_devops.defaultStorageClass(default_storage_classes_rwo)) }}" # if cpd 4.6 then we need RWX as SC, otherwise it will be RWO.
     cpd_service_block_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.defaultStorageClass(default_storage_classes_rwo) }}" # This var is only needed by WD
+- name: Default CPD Service Storage if not set by user (Cognos Analytics only)
```

Comment:
If Cognos uses only file storage class and does not need an specific file storage class then we probably dont need this code and we can live with the defaulted options? i.e ibmc-file-gold-gid ?